.bootstrap.mk

```diff
@@ -48,42 +48,39 @@ SHELL       := bash
 # Determine the root directory of our codebase and export it, this allows easy
 # file inclusion in both Bash and Make.
 override ROOT := $(shell path="$(CURDIR)"; while [[ "$${path}" != "/" \
- && ! -f "$${path}/.root.mk" ]]; do path="$${path%/*}"; done; echo "$${path}")
+ && ! -f "$${path}/.bootstrap.mk" ]]; do path="$${path%/*}"; done; echo "$${path}")
 
 # A generic help message that parses the available targets, and lists each one
 # that has a comment on the same line with a ## prefix.
 help: ## Display this help
-	readonly pad=25
-	targets() {
-		local targets=()
-		local pattern='[^:]+:[^#]*## .*'
-
-		for mk in "$$@"; do
-			while read -r line; do
-				if [[ "$${line}" =~ $${pattern} ]]; then
-					targets+=("$${line}")
-				fi
-			done < "$${mk}"
-		done
-
-		readarray -t targets < <(for t in "$${targets[@]}"; do echo "$$t"; done | sort -d)
-
-		for target in "$${targets[@]}"; do
-			printf "    \033[0;32m%-$${pad}s\033[0m %s\n" "$${target%%:*}" "$${target##*## }"
-		done
-	}
-
-	echo
-	echo -e "\033[0;33mUsage:\033[0m
-		make [flags...] [target...] [options...]
-
-	\033[0;33mFlags:\033[0m
-		See \033[1;30mmake --help\033[0m
-
-	\033[0;33mTargets:\033[0m"
-
-	targets $(MAKEFILE_LIST)
-
+	readonly pad=25; \
```

Comment:
@msiebeneicher what's with all the backslashes?
That seems unnecessary seeing that we defined `.ONESHELL:` above.

Reply:
.ONESHELL is available since make version 3.82 (see https://stackoverflow.com/questions/32153034/oneshell-not-working-properly-in-makefile). This changes make make-bootstrap also compatible for older versions. Tested with `GNU Make 3.81`.

Reply:
Make 3.81 is from 2006, 3.82 from 2010. I don't see the need to support this.
What kind of dinosaur-age machines are you trying to run this on? 😜

Reply:
not trying - still have to do ;P
I have my fork in production ..